pbsmrtpipe/loader.py

```
@@ -90,7 +66,28 @@ def _load_all_pb_static_tasks(module_name, registered_tasks_d, filter_filename_f
     return _REGISTERED_STATIC_TASKS
 
 
-def load_all_pb_tool_contracts():
+def _get_env_path_if_defined(env_var):
+    """Get a Config env variable directory, or return None"""
+    path = os.getenv(env_var)
+    if path is not None:
+        if os.path.isdir(path):
+            return os.path.abspath(path)
+        else:
+            warnings.warn("Skipping loading contracts from {e} Enable to find {p}".format(e=env_var, p=path))
```

```python
def _get_env_paths(env_var):
  """An empty string implies CWD.
  """
  if env_var in os.environ:
    ret = [os.path.abspath(path) for path in os[env_var].split(':')]
  else:
    ret = []
  log.info('%r -> %r' %(env_var, ret))
  return ret
...
tc_paths = _get_env_paths(...)
for tc_path in tc_paths:
  masks = load(tc_path)
  update(masks)
```
* Fewer if-statements.
  * Lower complexity.
* Consistent data-types.
  * No `None` alternative.
* Still easy to debug.
  * The log is crystal-clear, and a missing directory could be diagnosed trivially.
* Less redundant validation.
  * Let the loader fail on missing dir if that's the right behavior.
  * Let Python raise exceptions for you, at least at start-up.
* Does not bother user with warnings.
  * "Should I be worried?"